### DISCUSSION TITLE
A new cinnamon menu

### DISCUSSION THREAD

**Discussion by hsbasu** [linuxmint/discussions#6] [Ideas]
The current cinnamon main menu is adequately ergonomic. However, I have felt that the application menu provided in Deepin desktop is more intuitive. It offers both compact design and touch-friendly fullscreen modes. Can we add those feature in the current main menu?

**Comment by rcalixte** [comment]
@hsbasu Is it fair to assume you're referring to the feature highlighted in [this YouTube video](https://youtu.be/29onMCIbbXc?t=735)?

**Reply by hsbasu** [reply]
Yes!! This one. I don't know whether this is possible or not. @mtwebster should be able to clarify more. As far as I know he was working on something similar few years back.

**Reply by clefebvre** [reply]
Can you detail which exact features you like (don't hesitate to use screenshots) and how you're planning on adding them to the Cinnamon menu?

**Reply by hsbasu** [reply]
@clefebvre I confess, I unfortunately don't have the ability to code the whole menu. But I certainly can list the features that I think would be useful. For starters, I would suggest checking out the YouTube link @rcalixte posted.

**Reply by rcalixte** [reply]
I took the liberty of exporting the relevant clip and uploading it here for easier consumption.

https://user-images.githubusercontent.com/10281587/236549628-3050b2a7-af04-4259-9c5a-acaa1f11cc77.mp4

**Reply by rcalixte** [reply]
If no one else is interested in taking this up, I wouldn't mind giving it a shot before Cinnamon 6.0 (skipping the upcoming 5.8 release).

**Reply by hsbasu** [reply]
@rcalixte it would be much appreciated if you could take it up and give it a shot.

**Reply by clefebvre** [reply]
I watched the video. I like the way it looks but I think it's worth debating the pros and cons and breaking down the ideas first before rushing into an implementation.

**Reply by hsbasu** [reply]
I second that.

**Reply by hsbasu** [reply]
One of the important issues is that this start menu is for KDE. So, how much can be imitated is also questionable. Whereas In cinnamon we use gtk toolkit.

**Reply by Secret-chest** [reply]
> I watched the video. I like the way it looks but I think it's worth debating the pros and cons and breaking down the ideas first before rushing into an implementation.

The classic menu should be kept as a separate applet then. I hate how you need 2 clicks to browse by category.

**Reply by Secret-chest** [reply]
I also hate this glasmorphism personally. And the GNOME overwiew should be a new applet.

**Comment by JosephMcc** [comment]
One thing to keep in mind when working on the menu is ease of use and convenience. It's something where Cinnamon's current menu excels. Even in the video thumbnail above you can see an immediate issue. That is having the applications list and categories on two separate "pages". Having used Cinnamon's menu for a long time and then using desktops that have a menu system like that is an annoyance. Cinnamon's has everything right there and available just by hovering the mouse around which is quite nice.

**Reply by Secret-chest** [reply]
Agree. That menu is straight from hell, even worse than the Windows 95 one. Not even cascading menus.

**Comment by goyalyashpal** [comment]
try cinnamenu, it's awesome 😇

especially the `e smiley` emoji picker too 

Cinnamenu
https://cinnamon-spices.linuxmint.com/applets/view/322

**Reply by cplittlfield** [reply]
I like the menu as it currently exists.  Please don't "fix" it just for the sake of changing things around.  I like the prototype Joseph posted on the Web site, though some people are complaining about seeing the username and icon.  I generally find apps by using the search feature, if that is useful to know.

**Reply by goyalyashpal** [reply]
hi!  as you replied to my comment particularly, my aim was not to replace the default one with this "applet".

rather, this was a suggestion to the creator of this discussion that if per doesn't like the default one, then per may want to try out this feature rich add-in replacement.  

this is the beauty of FOSS things, that one may freely mix and match components without being dependent on the owner.

**Reply by cplittlfield** [reply]
I'm very sorry, that wasn't directed at you in particular.  I am still figuring out the Github commenting system.

But you are absolutely right.  I agree with you.  What I was actually talking about is taking a functional piece of software that has a reasonably effective interface and replacing it with something obtuse and incomprehensible, solely because the current interface "looks old" and doesn't follow the current fashions.

I'm not arguing against an _improved_ interface or _better_ functionality, mind you.

**Reply by Jopp-gh** [reply]
Cinnamon menu is very useful indeed, nicely designed and I could not be without, sometimes.

But for quick access, [Ulauncher](https://ulauncher.io/) is definitely the better choice, because more flexible, faster and focused.
